### PR TITLE
Add booking by simple token

### LINKED ISSUE #489: When user logs in via token in MMB, he isn't logged into SmartFAQ
Video: https://monosnap.com/file/JVnQ3kk46pbaFZjWNwuc4dHGSzUnHF

### PR DESCRIPTION
**Pull request by jalbertsr**
close #489 

Merge only after closing https://github.com/kiwicom/graphql/pull/289

<br/><url>LiveURL: https://smartfaq-489-MMB-via-email-token.surge.sh</url>

### CODE REVIEW

**Comment by jaroslav-kubicek** [comment]
conflicts, cc @jalbertsr

File: src/helpers/UserStatus.js [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
   children?: React.Node,
 };
 
-const RequireUserCriteria = (user_criteria: (user: User) => boolean) =>
-  withUser((props: Props) => {
-    if (!user_criteria(props.user)) return null;
-    return props.children;
-  });
+const RequireUserCriteria = (
+  user_criteria: (user: User | ?string) => boolean,
```
Comment (jaroslav-kubicek) [inline src/helpers/UserStatus.js]: can we use camelCase here?

File: src/helpers/UserStatus.js [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
+) => {
+  return withSimpleToken(
+    withUser((props: Props) => {
+      if (!user_criteria(props.user) && !user_criteria(props.simpleToken))
```
Comment (jaroslav-kubicek) [inline src/helpers/UserStatus.js]: I don't get this part, is even `user_criteria` needed?

looks like `if (!props.user && !user.simpleToken)` would do the job

**Comment by jaroslav-kubicek** [comment]
is this still WIP?

**Comment by jalbertsr** [comment]
@jaroslav-kubicek hmmm I just realized that I also need to add this new query to the boarding pass button, like here we also use `booking` https://github.com/kiwicom/smart-faq/blob/master/src/StaticFAQ/FAQExtraInfo/BoardingPassesInfo/index.js#L51

File: .eslintrc [hunk 3]
```diff
@@ -90,6 +90,13 @@
       "error",
       "always"
     ],
+    "flowtype/space-after-type-colon": [
+      2,
+      "always",
+      {
+        "allowLineBreak": true
```
Comment (jalbertsr) [inline .eslintrc]: I needed to add `allowLineBreak` to be available to do 
https://github.com/kiwicom/smart-faq/pull/560/files#diff-7675e72b59cb13d5f92a71de8870e806R24

File: src/helpers/UserStatus.js [hunk 4]
```diff
@@ -1,20 +1,25 @@
...
+  simpleToken: ?string,
   children?: React.Node,
 };
 
-const RequireUserCriteria = (user_criteria: (user: User) => boolean) =>
-  withUser((props: Props) => {
-    if (!user_criteria(props.user)) return null;
-    return props.children;
-  });
+const RequireUserCriteria = () =>
```
Comment (jaroslav-kubicek) [inline src/helpers/UserStatus.js]: it didn't need to be function returning function if it's used just as `LoggedIn: RequireUserCriteria`

Comment (jaroslav-kubicek) [reply src/helpers/UserStatus.js]: ...you can fix it in some following PR, no big deal

**Pull request merged**